jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RedirectRegexRule.java

```diff
@@ -84,6 +107,26 @@ protected Handler apply(Handler input, Matcher matcher) throws IOException
             protected boolean handle(Response response, Callback callback)
             {
                 String target = matcher.replaceAll(getLocation());
+
+                if (isAddQueries())
```

Comment:
```suggestion
                if (isAddQueries() && input.getHttpURI().getQuery() != null)
```

or maybe a StringUtil.isNotBlank test?